R/makeTreeSummarizedExperimentFromPhyloseq.R

```diff
@@ -1,90 +0,0 @@
-#' Coerce a \code{phyloseq} object to a \code{TreeSummarizedExperiment}
-#'
-#' \code{makeTreeSEFromPhyloseq} converts \code{phyloseq}
-#' objects into \code{TreeSummarizedExperiment} objects.
-#'
-#' All data stored in a \code{phyloseq} object is transferred.
-#'
-#' @param obj a \code{phyloseq} object
-#'
-#' @return An object of class \code{TreeSummarizedExperiment}
-#'
-#' @importFrom S4Vectors SimpleList DataFrame make_zero_col_DFrame
-#' @importFrom SummarizedExperiment colData colData<-
-#'
-#' @export
-#'
-#' @name makeTreeSEFromPhyloseq
-#' @seealso
-#' \code{\link[=makeTreeSEFromBiom]{makeTreeSEFromBiom}}
-#' \code{\link[=makeTreeSEFromDADA2]{makeTreeSEFromDADA2}}
-#' \code{\link[=importQIIME2]{importQIIME2}}
-#' \code{\link[=importMothur]{importMothur}}
-#'
-#' @examples
-#' if (requireNamespace("phyloseq")) {
-#'     data(GlobalPatterns, package="phyloseq")
-#'     makeTreeSEFromPhyloseq(GlobalPatterns)
-#'     data(enterotype, package="phyloseq")
-#'     makeTreeSEFromPhyloseq(enterotype)
-#'     data(esophagus, package="phyloseq")
-#'     makeTreeSEFromPhyloseq(esophagus)
-#' }
-makeTreeSEFromPhyloseq <- function(obj) {
-    # input check
-    .require_package("phyloseq")
-    if(!is(obj,"phyloseq")){
-        stop("'obj' must be a 'phyloseq' object")
-    }
-    #
-    # Get the assay
-    counts <- obj@otu_table@.Data
-    # Check the orientation, and transpose if necessary
-    if( !obj@otu_table@taxa_are_rows ){
-        counts <- t(counts)
-    }
-    # Create a list of assays
-    assays <- SimpleList(counts = counts)
-    
-    if(!is.null(obj@tax_table@.Data)){
-        rowData <- DataFrame(data.frame(obj@tax_table@.Data))
-    } else{
-        rowData <- S4Vectors::make_zero_col_DFrame(nrow(assays$counts))
-        rownames(rowData) <- rownames(assays$counts)
-    }
-    if(!is.null(obj@sam_data)){
-        colData <- DataFrame(data.frame(obj@sam_data))
-    } else{
-        colData <- S4Vectors::make_zero_col_DFrame(ncol(assays$counts))
-        rownames(colData) <- colnames(assays$counts)
-    }
-    if(!is.null(obj@phy_tree)){
-        rowTree <- obj@phy_tree
-    } else {
-        rowTree <- NULL
-    }
-    if (!is.null(obj@refseq)) {
-        referenceSeq <- obj@refseq
-    } else {
-        referenceSeq <- NULL
-    }
-    TreeSummarizedExperiment(assays = assays,
-                            rowData = rowData,
```

Comment:
Not sure, if the link is just broken between files, but it seems now that you have copy.-pasted code to another file. Instead, it should only look like that the name has changed.

Also, as this is not anymore big change (only the name is changed and documentation is moved under common page), I am wondering if you should make new PR with new branch if that is easier.

We have parameter naming PR that will be soon merged, and it affects also these files. It was good that we explored the possibility of common convert function even though it will not be implemented.

I am wondering what is the easiest approach.

Reply:
Yes I think it may be easier to open a new PR because I have probably made some unnecessary changes in this PR when I tried to implement `convert` generic function.

Reply:
Should I make new methods for the functions that do not have one (e.g. `makeTreeSEFromDADA2`) ?

Also should we keep both names makeTreeSE* and makeTreeSummarizedExperiment* for every converters or not?

Reply:
Now this bigger PR was merged https://github.com/microbiome/mia/pull/582, so you can continue with this.

Those names will be deprecated. All the functions that were available for user must be deprecated and added to deprecate.R file.

Each functionality will have only one function that is convertTo* or convertFrom*